node_modules/gh-users/tests/test-users.js

```diff
@@ -522,4 +523,137 @@ describe('Users', function() {
             });
         });
     });
+
+    describe('Changing a user\'s password', function() {
+
+        /**
+         * Test that verifies that a user's password can be updated
+         */
+        it('verify a user\'s password can be updated', function(callback) {
+            TestsUtil.generateTestUsers(global.tests.apps.cam2014, 1, false, function(simon) {
+
+                UsersTestsUtil.assertChangePassword(simon.client, simon.profile.id, 'a new password', simon.client.options.password, function(newClient) {
+                    return callback();
+                });
+            });
+        });
+
+        /**
+         * Test that verifies validation when changing a user's password
+         */
+        it('verify validation when changing a user\'s password', function(callback) {
+            TestsUtil.generateTestUsers(global.tests.apps.cam2014, 1, false, function(simon) {
+
+                var newPassword = 'new password';
+                var oldPassword = simon.client.options.password;
+
+                // Invalid user id
+                UsersTestsUtil.assertChangePasswordFails(simon.client, 'Not a number', newPassword, oldPassword, 400, function() {
+                    // Unknown user id
+                    UsersTestsUtil.assertChangePasswordFails(simon.client, -1, newPassword, oldPassword, 404, function() {
+
+                        // Missing new password
+                        UsersTestsUtil.assertChangePasswordFails(simon.client, simon.profile.id, null, oldPassword, 400, function() {
+
+                            // Too short new password
+                            UsersTestsUtil.assertChangePasswordFails(simon.client, simon.profile.id, 's', oldPassword, 400, function() {
+
+                                // Missing old password
+                                UsersTestsUtil.assertChangePasswordFails(simon.client, simon.profile.id, newPassword, null, 400, function() {
+
+                                    // Too short old password
+                                    UsersTestsUtil.assertChangePasswordFails(simon.client, simon.profile.id, newPassword, 's', 400, function() {
+
+                                        // Verify administrators don't need to provide an old password
+                                        UsersTestsUtil.assertChangePassword(global.tests.admins.cam2014.client, simon.profile.id, 'a new password', null, function(newClient) {
```

Comment:
Another potential issue is that compromised OAuth clients would be able to hijack accounts.